File: clients/typescript/src/models/SearchHit.ts
/* tslint:disable */
/* eslint-disable */

import { mapValues } from '../runtime';
import type { MediaSpan } from './MediaSpan';
import {
    MediaSpanFromJSON,
    MediaSpanFromJSONTyped,
    MediaSpanToJSON,
    MediaSpanToJSONTyped,
} from './MediaSpan';

export interface SearchHit {

    blobId: number;

    caption?: string | null;

    customerId?: string | null;

    duplicateCount?: number;

    duplicates?: Array<number>;

    geoLabel?: string | null;

    headline?: string | null;

    height?: number | null;

    kind?: SearchHitKindEnum | null;

    mediaRef?: string | null;

    metadata?: { [key: string]: any; } | null;

    score: number;

    scoreType: SearchHitScoreTypeEnum;

    seekMs?: number | null;

    span?: MediaSpan | null;

    spanHeadline?: string | null;

    tags?: Array<string>;

    width?: number | null;
}

export const SearchHitKindEnum = {
    Image: 'image',
    Video: 'video',
    Audio: 'audio',
    Pdf: 'pdf',
    Doc: 'doc',
    Note: 'note',
} as const;
export type SearchHitKindEnum = typeof SearchHitKindEnum[keyof typeof SearchHitKindEnum];

export const SearchHitScoreTypeEnum = {
    Cosine: 'cosine',
    Cosinelex: 'cosine+lex',
    Rrf: 'rrf',
    Rerank: 'rerank',
} as const;
export type SearchHitScoreTypeEnum = typeof SearchHitScoreTypeEnum[keyof typeof SearchHitScoreTypeEnum];

export function instanceOfSearchHit(value: object): value is SearchHit {
    if ((!('blobId' in (value as Record<string, any>)) && !('blob_id' in (value as Record<string, any>))) || ((value as Record<string, any>)['blobId'] === undefined && (value as Record<string, any>)['blob_id'] === undefined)) return false;
    if (!('score' in value) || value['score'] === undefined) return false;
    if ((!('scoreType' in (value as Record<string, any>)) && !('score_type' in (value as Record<string, any>))) || ((value as Record<string, any>)['scoreType'] === undefined && (value as Record<string, any>)['score_type'] === undefined)) return false;
    return true;
}

export function SearchHitFromJSON(json: any): SearchHit {
    return SearchHitFromJSONTyped(json, false);
}

export function SearchHitFromJSONTyped(json: any, ignoreDiscriminator: boolean): SearchHit {
    if (json == null) {
        return json;
    }
    return {

        'blobId': json['blob_id'],
        'caption': json['caption'] === undefined ? undefined : json['caption'] === null ? null : json['caption'],
        'customerId': json['customer_id'] === undefined ? undefined : json['customer_id'] === null ? null : json['customer_id'],
        'duplicateCount': json['duplicate_count'] == null ? undefined : json['duplicate_count'],
        'duplicates': json['duplicates'] == null ? undefined : json['duplicates'],
        'geoLabel': json['geo_label'] === undefined ? undefined : json['geo_label'] === null ? null : json['geo_label'],
        'headline': json['headline'] === undefined ? undefined : json['headline'] === null ? null : json['headline'],
        'height': json['height'] === undefined ? undefined : json['height'] === null ? null : json['height'],
        'kind': json['kind'] === undefined ? undefined : json['kind'] === null ? null : json['kind'],
        'mediaRef': json['media_ref'] === undefined ? undefined : json['media_ref'] === null ? null : json['media_ref'],
        'metadata': json['metadata'] === undefined ? undefined : json['metadata'] === null ? null : json['metadata'],
        'score': json['score'],
        'scoreType': json['score_type'],
        'seekMs': json['seek_ms'] === undefined ? undefined : json['seek_ms'] === null ? null : json['seek_ms'],
        'span': json['span'] === undefined ? undefined : json['span'] === null ? null : MediaSpanFromJSON(json['span']),
        'spanHeadline': json['span_headline'] === undefined ? undefined : json['span_headline'] === null ? null : json['span_headline'],
        'tags': json['tags'] == null ? undefined : json['tags'],
        'width': json['width'] === undefined ? undefined : json['width'] === null ? null : json['width'],
    };
}

export function SearchHitToJSON(json: any): SearchHit {
    return SearchHitToJSONTyped(json, false);
}

export function SearchHitToJSONTyped(value?: SearchHit | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {

        'blob_id': value['blobId'],
        'caption': value['caption'],
        'customer_id': value['customerId'],
        'duplicate_count': value['duplicateCount'],
        'duplicates': value['duplicates'],
        'geo_label': value['geoLabel'],
        'headline': value['headline'],
        'height': value['height'],
        'kind': value['kind'],
        'media_ref': value['mediaRef'],
        'metadata': value['metadata'],
        'score': value['score'],
        'score_type': value['scoreType'],
        'seek_ms': value['seekMs'],
        'span': MediaSpanToJSON(value['span']),
        'span_headline': value['spanHeadline'],
        'tags': value['tags'],
        'width': value['width'],
    };
}

File: clients/rust/src/models/search_hit.rs
use crate::models;
use serde::{Deserialize, Serialize};

#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
pub struct SearchHit {
    #[serde(rename = "blob_id")]
    pub blob_id: i32,
    #[serde(rename = "caption", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub caption: Option<Option<String>>,
    #[serde(rename = "customer_id", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub customer_id: Option<Option<String>>,
    #[serde(rename = "duplicate_count", skip_serializing_if = "Option::is_none")]
    pub duplicate_count: Option<i32>,
    #[serde(rename = "duplicates", skip_serializing_if = "Option::is_none")]
    pub duplicates: Option<Vec<i32>>,
    #[serde(rename = "geo_label", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub geo_label: Option<Option<String>>,
    #[serde(rename = "headline", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub headline: Option<Option<String>>,
    #[serde(rename = "height", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub height: Option<Option<i32>>,
    #[serde(rename = "kind", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub kind: Option<Option<Kind>>,
    #[serde(rename = "media_ref", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub media_ref: Option<Option<String>>,
    #[serde(rename = "metadata", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub metadata: Option<Option<std::collections::HashMap<String, serde_json::Value>>>,
    #[serde(rename = "score")]
    pub score: f64,
    #[serde(rename = "score_type")]
    pub score_type: ScoreType,
    #[serde(rename = "seek_ms", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub seek_ms: Option<Option<i32>>,
    #[serde(rename = "span", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub span: Option<Option<Box<models::MediaSpan>>>,
    #[serde(rename = "span_headline", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub span_headline: Option<Option<String>>,
    #[serde(rename = "tags", skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,
    #[serde(rename = "width", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub width: Option<Option<i32>>,
}

impl SearchHit {
    pub fn new(blob_id: i32, score: f64, score_type: ScoreType) -> SearchHit {
        SearchHit {
            blob_id,
            caption: None,
            customer_id: None,
            duplicate_count: None,
            duplicates: None,
            geo_label: None,
            headline: None,
            height: None,
            kind: None,
            media_ref: None,
            metadata: None,
            score,
            score_type,
            seek_ms: None,
            span: None,
            span_headline: None,
            tags: None,
            width: None,
        }
    }
}
#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum Kind {
    #[serde(rename = "image")]
    Image,
    #[serde(rename = "video")]
    Video,
    #[serde(rename = "audio")]
    Audio,
    #[serde(rename = "pdf")]
    Pdf,
    #[serde(rename = "doc")]
    Doc,
    #[serde(rename = "note")]
    Note,
}

impl Default for Kind {
    fn default() -> Kind {
        Self::Image
    }
}
#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum ScoreType {
    #[serde(rename = "cosine")]
    Cosine,
    #[serde(rename = "cosine+lex")]
    CosinePlusLex,
    #[serde(rename = "rrf")]
    Rrf,
    #[serde(rename = "rerank")]
    Rerank,
}

impl Default for ScoreType {
    fn default() -> ScoreType {
        Self::Cosine
    }
}

